src/app/commissioner_app.cpp

```diff
@@ -112,6 +112,14 @@ Error CommissionerApp::Start(std::string &      aExistingCommissionerId,
 void CommissionerApp::Stop()
 {
     IgnoreError(mCommissioner->Resign());
+
+    mJoiners.clear();
```

Comment:
Are we allowing `CommissionerApp` to be reused for multiple times (`Start`, `Stop` for multiple times) ? 
One design choice is to not allow a stopped `CommissionerApp` to be restarted. In this way we could simplify state management for `CommissionerApp`?

Reply:
Yes, we allow `CommissionerApp` to be restarted for multiple times. Currently, `CommissionerApp` is for just commissioner CLI. and methods `Start()`, `Stop` are mapped to CLI command `start ...`, `stop`.